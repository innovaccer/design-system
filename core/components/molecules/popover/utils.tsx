import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
