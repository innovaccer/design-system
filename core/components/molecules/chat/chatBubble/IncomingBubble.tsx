import * as React from 'react';
import { AccentAppearance } from '@/common.type';
import styles from '@css/components/chatBubble.module.css';
import { Text, Avatar, Row, Column } from '@/index';
import classNames from 'classnames';
import { BaseProps } from '@/utils/types';

export interface ChatAvatarProps {
  appearance?: AccentAppearance;
  firstName?: string;
  lastName?: string;
  role?: string;
  tabIndex?: number;
  icon?: React.ReactNode;
  image?: React.ReactNode;
}

export interface IncomingOptionProps extends BaseProps {
  children?: React.ReactNode;
  time?: string | React.ReactText;
  metaData?: string;
  actionBar?: () => JSX.Element;
  urgentMessage?: () => JSX.Element;
  avatarData?: ChatAvatarProps;
  showAvatar?: boolean;
}

const MetaSeparator = () => (
  <span className={styles['ChatBubble-separator']} data-test="DesignSystem-IncomingChatBubble-Separator" />
);

export const IncomingBubble = (props: IncomingOptionProps) => {
  const { time, metaData, urgentMessage, avatarData = {}, showAvatar, children, actionBar, ...rest } = props;
  const { image, icon, firstName, lastName } = avatarData;
  const fullName = `${firstName ? firstName : ''} ${lastName ? lastName : ''}`;

  const [showActionBar, setShowActionBar] = React.useState(false);

  const showMetaRow = firstName || lastName || time || metaData || urgentMessage;

  const metaDataClass = classNames({
    ['d-flex align-items-center mb-3']: true,
    [styles['ChatBubble-metaData--incoming']]: showAvatar,
  });

  return (
    <div
      data-test="DesignSystem-ChatBubble-IncomingWrapper"
      {...rest}
      role="group"
      aria-labelledby="chat-bubble-header"
    >
      {showMetaRow && (
        <Row className={metaDataClass} data-test="DesignSystem-IncomingChatBubble-MetaDataWrapper">
          {[
            fullName && (firstName || lastName) && (
              <Text key="fullName" weight="medium" size="small" data-test="DesignSystem-IncomingChatBubble-Name">
                {fullName}
              </Text>
            ),
            time && (
              <Text
                key="time"
                appearance="subtle"
                weight="medium"
                size="small"
                data-test="DesignSystem-IncomingChatBubble-Time"
                aria-label={`Time: ${time}`}
              >
                {time}
              </Text>
            ),
            metaData && (
              <Text
                key="metaData"
                appearance="subtle"
                weight="medium"
                size="small"
                data-test="DesignSystem-IncomingChatBubble-MetaData"
                aria-label={metaData}
              >
                {metaData}
              </Text>
            ),
            urgentMessage && (
              <div
                key="urgentMessage"
                data-test="DesignSystem-IncomingChatBubble-UrgentMessage"
                role="alert"
                aria-live="assertive"
              >
                {urgentMessage()}
              </div>
            ),
          ]
            .filter(Boolean)
            .map((element, index, array) => (
              <React.Fragment key={index}>
                {element}
                {index < array.length - 1 && <MetaSeparator key={`separator-${index}`} />}
              </React.Fragment>
            ))}
        </Row>
      )}
      <Row
        onMouseEnter={() => setShowActionBar(true)}
        onMouseLeave={() => setShowActionBar(false)}
        data-test="DesignSystem-IncomingChatBubble-Wrapper"
      >
        <Column className={styles['ChatBubble-boxWrapper']} data-test="DesignSystem-IncomingChatBubble-ChatBoxWrapper">
          {showAvatar && (
            <Avatar
              data-test="DesignSystem-IncomingChatBubble-Avatar"
              className="mr-4"
              {...avatarData}
              withTooltip={false}
              aria-label={`Avatar of ${fullName}`}
            >
              {image || icon}
            </Avatar>
          )}
          <div className={styles['ChatBubble-box--incoming']} data-test="DesignSystem-IncomingChatBubble-ChatBox">
            {children}
          </div>
        </Column>
        <Column className={styles['ChatBubble-actionBarWrapper']}>
          {actionBar && showActionBar && (
            <div
              data-test="DesignSystem-IncomingChatBubble-ActionBar"
              className={styles['ChatBubble-actionBar--incoming']}
              role="toolbar"
              aria-label="Action bar"
            >
              {actionBar()}
            </div>
          )}
        </Column>
      </Row>
    </div>
  );
};

IncomingBubble.displayName = 'IncomingBubble';
export default IncomingBubble;
