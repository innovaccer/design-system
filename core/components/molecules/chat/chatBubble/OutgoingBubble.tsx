import * as React from 'react';
import { Icon, Text, Row, Column } from '@/index';
import styles from '@css/components/chatBubble.module.css';
import classNames from 'classnames';
import { BaseProps } from '@/utils/types';

export interface OutgoingOptionProps extends BaseProps {
  metaData?: string;
  status?: boolean;
  failed?: boolean;
  children?: React.ReactNode;
  time?: string | React.ReactText;
  actionBar?: () => JSX.Element;
  urgentMessage?: () => JSX.Element;
  failedMessage?: () => JSX.Element;
}

const ChatBox = (props: { children: React.ReactNode }) => {
  const { children } = props;

  return (
    <div className={styles['ChatBubble-box--outgoing']} data-test="DesignSystem-OutgoingChatBubble-ChatBox">
      {children}
    </div>
  );
};

const MetaSeparator = () => (
  <span className={styles['ChatBubble-separator']} data-test="DesignSystem-OutgoingChatBubble-Separator" />
);

export const OutgoingBubble = (props: OutgoingOptionProps) => {
  const { children, status, actionBar, time, metaData, urgentMessage, failed, failedMessage, className, ...rest } =
    props;

  const [showActionBar, setShowActionBar] = React.useState(false);

  const chatBoxClass = classNames({
    [styles['ChatBubble-box']]: true,
    [styles['ChatBubble-box--noSuccess']]: !status,
  });

  const chatBoxWrapperClass = classNames('d-flex', 'align-items-end', 'w-100', className);

  const showMetaRow = time || metaData || urgentMessage || (failed && failedMessage);

  const metaRowClass = classNames({
    ['d-flex justify-content-end align-items-center mb-3']: showMetaRow,
  });

  return (
    <div
      className={chatBoxWrapperClass}
      data-test="DesignSystem-ChatBubble-OutgoingWrapper"
      {...rest}
      role="group"
      aria-labelledby="chat-bubble-header"
    >
      <div className={chatBoxClass}>
        {showMetaRow && (
          <Row className={metaRowClass}>
            {failed && failedMessage && (
              <div data-test="DesignSystem-OutgoingChatBubble-FailedMessage" role="alert" aria-live="assertive">
                {failedMessage()}
              </div>
            )}
            {!failed && (
              <>
                {[
                  time && (
                    <Text
                      key="time"
                      appearance="subtle"
                      weight="medium"
                      size="small"
                      data-test="DesignSystem-OutgoingChatBubble-Time"
                      aria-label={`Time: ${time}`}
                    >
                      {time}
                    </Text>
                  ),
                  metaData && (
                    <Text
                      key="metaData"
                      appearance="subtle"
                      weight="medium"
                      size="small"
                      data-test="DesignSystem-OutgoingChatBubble-Metadata"
                      aria-label={metaData}
                    >
                      {metaData}
                    </Text>
                  ),
                  urgentMessage && (
                    <div
                      key="urgentMessage"
                      data-test="DesignSystem-OutgoingChatBubble-UrgentMessage"
                      role="alert"
                      aria-live="assertive"
                    >
                      {urgentMessage()}
                    </div>
                  ),
                ]
                  .filter(Boolean)
                  .map((element, index, array) => (
                    <React.Fragment key={index}>
                      {element}
                      {index < array.length - 1 && <MetaSeparator key={`separator-${index}`} />}
                    </React.Fragment>
                  ))}
              </>
            )}
          </Row>
        )}
        <Row
          className="d-flex justify-content-end"
          onMouseEnter={() => setShowActionBar(true)}
          onMouseLeave={() => setShowActionBar(false)}
          data-test="DesignSystem-OutgoingChatBubble-Wrapper"
        >
          <Column className={styles['ChatBubble-actionBarWrapper']}>
            {actionBar && showActionBar && (
              <div
                className={styles['ChatBubble-actionBar--outgoing']}
                data-test="DesignSystem-OutgoingChatBubble-ActionBar"
                role="toolbar"
                aria-label="Action bar"
              >
                {actionBar()}
              </div>
            )}
          </Column>

          <Column className={styles['ChatBubble-boxWrapper']} data-test="DesignSystem-OutgoingChatBubble-BoxWrapper">
            <ChatBox>{children}</ChatBox>
          </Column>
        </Row>
      </div>

      {status && !failed && (
        <Icon
          name="check_circle"
          appearance="success"
          type="rounded"
          className="ml-3"
          data-test="DesignSystem-OutgoingChatBubble-Status"
          aria-label="Message sent successfully"
        />
      )}
    </div>
  );
};

OutgoingBubble.displayName = 'OutgoingBubble';
export default OutgoingBubble;
