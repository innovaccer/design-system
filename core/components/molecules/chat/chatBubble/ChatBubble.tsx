import * as React from 'react';
import { IncomingBubble, IncomingOptionProps } from './IncomingBubble';
import { OutgoingBubble, OutgoingOptionProps } from './OutgoingBubble';
import { BaseProps } from '@/utils/types';

export type ChatBubbleType = 'incoming' | 'outgoing';

export interface ChatBubbleProps extends BaseProps {
  /**
   * Type of ChatBubble
   */
  type: ChatBubbleType;
  /**
   * Props for Incoming Chat Bubble Type
   *
   * **`incomingOptions` are only applicable when `type` is `incoming`
   *
   * <pre style="font-family: monospace; font-size: 13px; background: #f8f8f8">
   * IncomingOptionProps: {
   *   children?: React.ReactNode;
   *   time?: string | React.ReactText;
   *   metaData?: string;
   *   actionBar?: () => JSX.Element;
   *   urgentMessage?: () => JSX.Element;
   *   avatarData?: ChatAvatarProps;
   *   showAvatar?: boolean;
   * }
   *
   * ChatAvatarProps: {
   *  appearance?: AvatarAppearance;
   *  firstName?: string;
   *  lastName?: string;
   *  role?: string;
   *  tabIndex?: number;
   *  icon?: React.ReactNode;
   *  image?: React.ReactNode;
   * }
   * </pre>
   */
  incomingOptions?: IncomingOptionProps;
  /**
   * Props for Outgoing Chat Bubble Type
   * **`outgoingOptions` are only applicable when `type` is `outgoing`
   * <pre style="font-family: monospace; font-size: 13px; background: #f8f8f8">
   * OutgoingOptionProps: {
   *  metaData?: string;
   *  status?: boolean;
   *  failed?: boolean;
   *  children?: React.ReactNode;
   *  time?: string | React.ReactText;
   *  actionBar?: () => JSX.Element;
   *  urgentMessage?: () => JSX.Element;
   *  failedMessage?: () => JSX.Element;
   * }
   * </pre>
   */
  outgoingOptions?: OutgoingOptionProps;
  /**
   * Elements to be rendered inside ChatBubble
   */
  children?: React.ReactNode;
}

export const ChatBubble = (props: ChatBubbleProps) => {
  const { type, incomingOptions, outgoingOptions, children, ...rest } = props;

  if (type === 'incoming') {
    return (
      <IncomingBubble {...incomingOptions} {...rest}>
        {children}
      </IncomingBubble>
    );
  }

  return (
    <OutgoingBubble {...outgoingOptions} {...rest}>
      {children}
    </OutgoingBubble>
  );
};

ChatBubble.displayName = 'ChatBubble';
export default ChatBubble;
