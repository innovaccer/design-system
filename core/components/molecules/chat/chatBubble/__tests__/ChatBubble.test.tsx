import * as React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Chat } from '@/index';
import { ChatBubbleProps as Props } from '../ChatBubble';
import { IncomingOptionProps } from '../IncomingBubble';
import { OutgoingOptionProps } from '../OutgoingBubble';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const chatMessage = 'Test Message';
const type = ['incoming', 'outgoing'];

const incomingOptions: IncomingOptionProps = {
  time: '1:00 AM',
  metaData: 'Meta Data',
  avatarData: {
    firstName: 'John',
    lastName: 'Doe',
  },
  showAvatar: true,
  urgentMessage: () => <div>Urgent</div>,
};

const outgoingOptions: OutgoingOptionProps = {
  status: true,
  time: '1:00 AM',
  metaData: 'Meta Data',
  urgentMessage: () => <div>Urgent</div>,
};

describe('ChatBubble component snapshot', () => {
  const mapper = {
    type: valueHelper(type, { required: true, iterate: true }),
    incomingOptions: valueHelper(incomingOptions, { required: true, iterate: false }),
    outgoingOptions: valueHelper(outgoingOptions, { required: true, iterate: false }),
    children: valueHelper(<div>{chatMessage}</div>, { required: true, iterate: false }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const tree = render(
        <Chat>
          <Chat.ChatBubble {...attr}>JD</Chat.ChatBubble>
        </Chat>
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('ChatBubble Component type: Incoming', () => {
  it('renders incoming chat bubble correctly', () => {
    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="incoming" incomingOptions={incomingOptions}>
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    expect(getByTestId('DesignSystem-IncomingChatBubble-Name')).toHaveTextContent('John Doe');
    expect(getByTestId('DesignSystem-IncomingChatBubble-Time')).toHaveTextContent('1:00 AM');
    expect(getByTestId('DesignSystem-IncomingChatBubble-MetaData')).toHaveTextContent('Meta Data');
    expect(getByTestId('DesignSystem-IncomingChatBubble-UrgentMessage')).toHaveTextContent('Urgent');
    expect(getByTestId('DesignSystem-IncomingChatBubble-ChatBox')).toHaveTextContent(chatMessage);

    expect(getByTestId('DesignSystem-IncomingChatBubble-Avatar')).toBeInTheDocument;
  });

  it('shows action bar on mouse enter and hides on mouse leave for incoming bubble', () => {
    const actionBar = () => <div>Action Bar</div>;
    const incomingOptions: IncomingOptionProps = {
      actionBar,
    };

    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="incoming" incomingOptions={incomingOptions}>
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    fireEvent.mouseEnter(getByTestId('DesignSystem-IncomingChatBubble-Wrapper'));
    expect(getByTestId('DesignSystem-IncomingChatBubble-ActionBar')).toBeInTheDocument();

    fireEvent.mouseLeave(getByTestId('DesignSystem-IncomingChatBubble-Wrapper'));
    expect(screen.queryByTestId('DesignSystem-IncomingChatBubble-ActionBar')).not.toBeInTheDocument();
  });

  it('applies custom className to incoming chat bubble', () => {
    const incomingOptions: IncomingOptionProps = {
      time: '1:00 AM',
      metaData: 'Meta Data',
    };

    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="incoming" incomingOptions={incomingOptions} className="custom-class">
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    expect(getByTestId('DesignSystem-ChatBubble-IncomingWrapper')).toHaveClass('custom-class');
  });
});

describe('ChatBubble Component type: Outgoing', () => {
  it('renders outgoing chat bubble correctly', () => {
    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="outgoing" outgoingOptions={outgoingOptions}>
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    expect(getByTestId('DesignSystem-OutgoingChatBubble-Time')).toHaveTextContent('1:00 AM');
    expect(getByTestId('DesignSystem-OutgoingChatBubble-Metadata')).toHaveTextContent('Meta Data');
    expect(getByTestId('DesignSystem-OutgoingChatBubble-UrgentMessage')).toHaveTextContent('Urgent');
    expect(getByTestId('DesignSystem-OutgoingChatBubble-BoxWrapper')).toHaveTextContent(chatMessage);
  });

  it('shows action bar on mouse enter and hides on mouse leave for outgoing bubble', () => {
    const actionBar = () => <div>Action Bar</div>;
    const outgoingOptions: OutgoingOptionProps = {
      actionBar,
    };

    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="outgoing" outgoingOptions={outgoingOptions}>
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    fireEvent.mouseEnter(getByTestId('DesignSystem-OutgoingChatBubble-Wrapper'));
    expect(getByTestId('DesignSystem-OutgoingChatBubble-ActionBar')).toBeInTheDocument();

    fireEvent.mouseLeave(getByTestId('DesignSystem-OutgoingChatBubble-Wrapper'));
    expect(screen.queryByTestId('DesignSystem-OutgoingChatBubble-ActionBar')).not.toBeInTheDocument();
  });

  it('renders outgoing chat bubble with failed message correctly', () => {
    const outgoingOptions: OutgoingOptionProps = {
      failed: true,
      failedMessage: () => <div>Failed to send</div>,
    };

    const { getByTestId } = render(
      <Chat>
        <Chat.ChatBubble type="outgoing" outgoingOptions={outgoingOptions}>
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    expect(getByTestId('DesignSystem-OutgoingChatBubble-FailedMessage')).toHaveTextContent('Failed to send');
    expect(getByTestId('DesignSystem-OutgoingChatBubble-ChatBox')).toHaveTextContent(chatMessage);
  });

  it('applies custom className to outgoing chat bubble', () => {
    const outgoingOptions: OutgoingOptionProps = {
      status: true,
      time: '1:00 AM',
      metaData: 'Meta Data',
    };

    render(
      <Chat>
        <Chat.ChatBubble type="outgoing" outgoingOptions={outgoingOptions} className="custom-class">
          {chatMessage}
        </Chat.ChatBubble>
      </Chat>
    );

    expect(screen.getByTestId('DesignSystem-ChatBubble-OutgoingWrapper')).toHaveClass('custom-class');
  });
});
