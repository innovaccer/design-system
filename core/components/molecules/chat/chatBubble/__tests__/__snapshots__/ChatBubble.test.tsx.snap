// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble component snapshot 
  type: "incoming", incomingOptions: {"time":"1:00 AM","metaData":"Meta Data","avatarData":{"firstName":"John","lastName":"Doe"},"showAvatar":true,"urgentMessage":"[Function]"}, outgoingOptions: {"status":true,"time":"1:00 AM","metaData":"Meta Data","urgentMessage":"[Function]"}, children: "[ReactNode]"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-test="DesignSystem-Chat"
      >
        <div
          aria-labelledby="chat-bubble-header"
          data-test="DesignSystem-ChatBubble-IncomingWrapper"
          role="group"
        >
          <div
            class="Row d-flex align-items-center mb-3 ChatBubble-metaData--incoming"
            data-test="DesignSystem-IncomingChatBubble-MetaDataWrapper"
          >
            <span
              class="Text Text--default Text--medium Text--small"
              data-test="DesignSystem-IncomingChatBubble-Name"
            >
              John Doe
            </span>
            <span
              class="ChatBubble-separator"
              data-test="DesignSystem-IncomingChatBubble-Separator"
            />
            <span
              aria-label="Time: 1:00 AM"
              class="Text Text--subtle Text--medium Text--small"
              data-test="DesignSystem-IncomingChatBubble-Time"
            >
              1:00 AM
            </span>
            <span
              class="ChatBubble-separator"
              data-test="DesignSystem-IncomingChatBubble-Separator"
            />
            <span
              aria-label="Meta Data"
              class="Text Text--subtle Text--medium Text--small"
              data-test="DesignSystem-IncomingChatBubble-MetaData"
            >
              Meta Data
            </span>
            <span
              class="ChatBubble-separator"
              data-test="DesignSystem-IncomingChatBubble-Separator"
            />
            <div
              aria-live="assertive"
              data-test="DesignSystem-IncomingChatBubble-UrgentMessage"
              role="alert"
            >
              <div>
                Urgent
              </div>
            </div>
          </div>
          <div
            class="Row"
            data-test="DesignSystem-IncomingChatBubble-Wrapper"
          >
            <div
              class="Col ChatBubble-boxWrapper"
              data-test="DesignSystem-IncomingChatBubble-ChatBoxWrapper"
            >
              <span
                class="position-relative d-inline-flex"
              >
                <span
                  class=""
                  data-test="DesignSystem-AvatarWrapper"
                  role="presentation"
                >
                  <span
                    class="Avatar Avatar--regular Avatar--accent1 Avatar--noInitials Avatar--default mr-4"
                    data-test="DesignSystem-IncomingChatBubble-Avatar"
                    tabindex="0"
                  >
                    <span
                      class="Text Text--white Text--medium Text--regular Avatar-content--regular Avatar-content"
                      data-test="DesignSystem-Text"
                    >
                      JD
                    </span>
                  </span>
                </span>
              </span>
              <div
                class="ChatBubble-box--incoming"
                data-test="DesignSystem-IncomingChatBubble-ChatBox"
              >
                JD
              </div>
            </div>
            <div
              class="Col ChatBubble-actionBarWrapper"
              data-test="DesignSystem-Column"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-test="DesignSystem-Chat"
    >
      <div
        aria-labelledby="chat-bubble-header"
        data-test="DesignSystem-ChatBubble-IncomingWrapper"
        role="group"
      >
        <div
          class="Row d-flex align-items-center mb-3 ChatBubble-metaData--incoming"
          data-test="DesignSystem-IncomingChatBubble-MetaDataWrapper"
        >
          <span
            class="Text Text--default Text--medium Text--small"
            data-test="DesignSystem-IncomingChatBubble-Name"
          >
            John Doe
          </span>
          <span
            class="ChatBubble-separator"
            data-test="DesignSystem-IncomingChatBubble-Separator"
          />
          <span
            aria-label="Time: 1:00 AM"
            class="Text Text--subtle Text--medium Text--small"
            data-test="DesignSystem-IncomingChatBubble-Time"
          >
            1:00 AM
          </span>
          <span
            class="ChatBubble-separator"
            data-test="DesignSystem-IncomingChatBubble-Separator"
          />
          <span
            aria-label="Meta Data"
            class="Text Text--subtle Text--medium Text--small"
            data-test="DesignSystem-IncomingChatBubble-MetaData"
          >
            Meta Data
          </span>
          <span
            class="ChatBubble-separator"
            data-test="DesignSystem-IncomingChatBubble-Separator"
          />
          <div
            aria-live="assertive"
            data-test="DesignSystem-IncomingChatBubble-UrgentMessage"
            role="alert"
          >
            <div>
              Urgent
            </div>
          </div>
        </div>
        <div
          class="Row"
          data-test="DesignSystem-IncomingChatBubble-Wrapper"
        >
          <div
            class="Col ChatBubble-boxWrapper"
            data-test="DesignSystem-IncomingChatBubble-ChatBoxWrapper"
          >
            <span
              class="position-relative d-inline-flex"
            >
              <span
                class=""
                data-test="DesignSystem-AvatarWrapper"
                role="presentation"
              >
                <span
                  class="Avatar Avatar--regular Avatar--accent1 Avatar--noInitials Avatar--default mr-4"
                  data-test="DesignSystem-IncomingChatBubble-Avatar"
                  tabindex="0"
                >
                  <span
                    class="Text Text--white Text--medium Text--regular Avatar-content--regular Avatar-content"
                    data-test="DesignSystem-Text"
                  >
                    JD
                  </span>
                </span>
              </span>
            </span>
            <div
              class="ChatBubble-box--incoming"
              data-test="DesignSystem-IncomingChatBubble-ChatBox"
            >
              JD
            </div>
          </div>
          <div
            class="Col ChatBubble-actionBarWrapper"
            data-test="DesignSystem-Column"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChatBubble component snapshot 
  type: "outgoing", incomingOptions: {"time":"1:00 AM","metaData":"Meta Data","avatarData":{"firstName":"John","lastName":"Doe"},"showAvatar":true,"urgentMessage":"[Function]"}, outgoingOptions: {"status":true,"time":"1:00 AM","metaData":"Meta Data","urgentMessage":"[Function]"}, children: "[ReactNode]"
 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-test="DesignSystem-Chat"
      >
        <div
          aria-labelledby="chat-bubble-header"
          class="d-flex align-items-end w-100"
          data-test="DesignSystem-ChatBubble-OutgoingWrapper"
          role="group"
        >
          <div
            class="ChatBubble-box"
          >
            <div
              class="Row d-flex justify-content-end align-items-center mb-3"
              data-test="DesignSystem-Row"
            >
              <span
                aria-label="Time: 1:00 AM"
                class="Text Text--subtle Text--medium Text--small"
                data-test="DesignSystem-OutgoingChatBubble-Time"
              >
                1:00 AM
              </span>
              <span
                class="ChatBubble-separator"
                data-test="DesignSystem-OutgoingChatBubble-Separator"
              />
              <span
                aria-label="Meta Data"
                class="Text Text--subtle Text--medium Text--small"
                data-test="DesignSystem-OutgoingChatBubble-Metadata"
              >
                Meta Data
              </span>
              <span
                class="ChatBubble-separator"
                data-test="DesignSystem-OutgoingChatBubble-Separator"
              />
              <div
                aria-live="assertive"
                data-test="DesignSystem-OutgoingChatBubble-UrgentMessage"
                role="alert"
              >
                <div>
                  Urgent
                </div>
              </div>
            </div>
            <div
              class="Row d-flex justify-content-end"
              data-test="DesignSystem-OutgoingChatBubble-Wrapper"
            >
              <div
                class="Col ChatBubble-actionBarWrapper"
                data-test="DesignSystem-Column"
              />
              <div
                class="Col ChatBubble-boxWrapper"
                data-test="DesignSystem-OutgoingChatBubble-BoxWrapper"
              >
                <div
                  class="ChatBubble-box--outgoing"
                  data-test="DesignSystem-OutgoingChatBubble-ChatBox"
                >
                  JD
                </div>
              </div>
            </div>
          </div>
          <i
            aria-label="Message sent successfully"
            class="material-symbols material-symbols-rounded Icon Icon--success ml-3"
            data-test="DesignSystem-OutgoingChatBubble-Status"
            role="button"
            style="font-size: 16px; width: 16px;"
          >
            check_circle
          </i>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-test="DesignSystem-Chat"
    >
      <div
        aria-labelledby="chat-bubble-header"
        class="d-flex align-items-end w-100"
        data-test="DesignSystem-ChatBubble-OutgoingWrapper"
        role="group"
      >
        <div
          class="ChatBubble-box"
        >
          <div
            class="Row d-flex justify-content-end align-items-center mb-3"
            data-test="DesignSystem-Row"
          >
            <span
              aria-label="Time: 1:00 AM"
              class="Text Text--subtle Text--medium Text--small"
              data-test="DesignSystem-OutgoingChatBubble-Time"
            >
              1:00 AM
            </span>
            <span
              class="ChatBubble-separator"
              data-test="DesignSystem-OutgoingChatBubble-Separator"
            />
            <span
              aria-label="Meta Data"
              class="Text Text--subtle Text--medium Text--small"
              data-test="DesignSystem-OutgoingChatBubble-Metadata"
            >
              Meta Data
            </span>
            <span
              class="ChatBubble-separator"
              data-test="DesignSystem-OutgoingChatBubble-Separator"
            />
            <div
              aria-live="assertive"
              data-test="DesignSystem-OutgoingChatBubble-UrgentMessage"
              role="alert"
            >
              <div>
                Urgent
              </div>
            </div>
          </div>
          <div
            class="Row d-flex justify-content-end"
            data-test="DesignSystem-OutgoingChatBubble-Wrapper"
          >
            <div
              class="Col ChatBubble-actionBarWrapper"
              data-test="DesignSystem-Column"
            />
            <div
              class="Col ChatBubble-boxWrapper"
              data-test="DesignSystem-OutgoingChatBubble-BoxWrapper"
            >
              <div
                class="ChatBubble-box--outgoing"
                data-test="DesignSystem-OutgoingChatBubble-ChatBox"
              >
                JD
              </div>
            </div>
          </div>
        </div>
        <i
          aria-label="Message sent successfully"
          class="material-symbols material-symbols-rounded Icon Icon--success ml-3"
          data-test="DesignSystem-OutgoingChatBubble-Status"
          role="button"
          style="font-size: 16px; width: 16px;"
        >
          check_circle
        </i>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
